slides/aml-16-032118-clustering-and-mixture-models/index.html

```diff
@@ -0,0 +1,756 @@
+- Remaining points are labeled outliers
+]
+]
+
+.right-column[
+.center[
+![:scale 80%](images/dbscan_algo.png)
+]
+]
+
+???
+
+---
+
+# Illustration
+
+.center[
+![:scale 70%](images/dbscan_illustration.png)
+]
+
+???
+
+---
+
+# Pros and Cons
+
+- Allows complex cluster shapes
+- Can detect outliers
+- Needs two parameters to adjust, eps is hard to
+pick (can be done based on number of clusters
+though).
+- Can learn arbitrary cluster shapes
+
+.center[
+![:scale 40%](images/dbscan_pro_con.png)
+]
+
+???
+
+---
+class: center, middle
+
+## (Gaussian) Mixture Models
+
+???
+
+---
+
+# Mixture Models
+
+- Generative model – find p(X).
+- Assume form of data generating process
+- Mixture model assumption:
+  - Data is mixture of small number of known distributions.
+  - Each mixture component distribution can be learned "simply"
+  - Each point comes from one particular component
+- We learn the component parameters and weights of
+components
+
+` $$ p(\mathbf{x}) = \sum_{j=1}^k \pi_k p_k(\mathbf{x} | \theta) $$ `
+
+???
+
+---
+
+# Gaussian Mixture Models
+
+- Each component is created by a Gaussian distribution.
+- There is a multinomial distribution over the components
+
+`$$ p(\mathbf{x}) = \sum_{j=1}^k \pi_k \mathcal{N}(\mathbf{x}, \mu_k, \Sigma_k) $$`
+
+- Non-convex optimization.
+- Alternately assign points to components and compute mean
+and variance (EM algorithm).
+- Initialized with K-means, random restarts
+
+???
+
+---
+
+`$$ p(\mathbf{x}) = \sum_{j=1}^k \pi_k \mathcal{N}(\mathbf{x}, \mu_k, \Sigma_k) $$`
+
+.center[
+![:scale 40%](images/gmm1.png)
+]
+
+.center[
+![:scale 40%](images/gmm2.png)
+]
+
+???
+
+---
+
+# Goals of Mixture Models
+
+- Create parametric density model.
+- Allows for testing how “likely” a new point is.
+- Clustering (each components is one cluster).
+- Feature Extraction
+
+.center[
+![:scale 80%](images/gmm2.png)
+]
+
+???
+
+---
+
+# Examples
+
+.wide-left-column[
+.smaller[
+```python
+from sklearn.mixture import GaussianMixture
+gmm = GaussianMixture(n_components=3)
+gmm.fit(X)
+print(gmm.means_)
+print(gmm.covariances_)
+```
+[[-2.286 -4.674] </br>
+ [-0.377  6.947] </br>
+ [ 8.685  5.206]] </br>
+[[[  6.651   2.066] </br>
+  [  2.066  13.759]] </br>
+ [[  5.467  -3.341] </br>
+  [ -3.341   4.666]] </br>
+ [[  1.481  -1.1  ] </br>
+  [ -1.1     4.191]]] </br>
+]
+]
+
+
+.narrow-right-column[
+.center[
+![:scale 80%](images/mm_examples_1.png)
+]
+.center[
+![:scale 80%](images/mm_examples_2.png)
+]
+]
+
+???
+
+---
+
+# Probability Estimates
+
+.left-column[
+.smaller[
+```python
+gmm.predict_proba(X)
+```
+```
+array([[ 1.   ,  0.   ,  0.   ],
+       [ 0.   ,  0.   ,  1.   ],
+       ...,
+       [ 1.   ,  0.   ,  0.   ],
+       [ 0.001,  0.999,  0.   ]])
+```
+.center[
+![:scale 60%](images/prob_est1.png)
+]
+]
+]
+
+.right-column[
+.smaller[
+```python
+# log probability under the model
+print(gmm.score(X))
+print(gmm.score_samples(X).shape)
+```
+</br>
+-5.50838313166 </br>
```

Comment:
again, please use triple backticks for outputs, also round to 3 decimal places.

Reply:
Done.